packages/client/src/createGeotimeSeriesProperty.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  GeotimeSeriesProperty,
+  TimeSeriesPoint,
+  TimeSeriesProperty,
+  TimeSeriesQuery,
+} from "@osdk/api";
+import { TimeseriesDurationMapping } from "@osdk/api";
+import type { TimeRange } from "@osdk/internal.foundry.core";
+import * as OntologiesV2 from "@osdk/internal.foundry.ontologiesv2";
+import type { MinimalClient } from "./MinimalClientContext.js";
+import {
+  iterateReadableStream,
+  parseStreamedResponse,
+} from "./util/streamutils.js";
+import { asyncIterPointsHelper, getTimeRange } from "./util/timeseriesUtils.js";
+
+export class GeotimeSeriesPropertyImpl<T extends GeoJSON.Point>
+  implements GeotimeSeriesProperty<T>
+{
+  #triplet: [string, any, string];
+  #client: MinimalClient;
+  lastFetchedValue: TimeSeriesPoint<T> | undefined;
+
+  constructor(
+    client: MinimalClient,
+    objectApiName: string,
+    primaryKey: any,
+    propertyName: string,
+  ) {
+    this.#client = client;
+    this.#triplet = [objectApiName, primaryKey, propertyName];
+  }
+
+  public async getLatestValue() {
+    const latestPointPromise = OntologiesV2.TimeSeriesValueBankProperties
+      .getLatestValue(
+        this.#client,
+        await this.#client.ontologyRid,
+        ...this.#triplet,
+      ) as Promise<TimeSeriesPoint<T>>;
```

Comment:
Just because `as` COULD cause issues, I wonder if its better to just set the return type of the function to `Promise<TimeSeriesPoint<T>>` instead? If its not assignable it will error.

Reply:
Do we run any risk of `getLatestValue` later returning additional information we didn't intend?

Reply:
I can change the return type, and no I don't forsee atleast adding anything new like that. Like it will always be a point of {time, value:GEOJSON of some form}

Reply:
If we need something drastically different, I think it'd just have to be a new endpoint